src/rgw/rgw_op.cc

```diff
@@ -6706,21 +6703,10 @@ void RGWCompleteMultipart::execute(optional_yield y)
     return;
   }
 
-  // remove the upload meta object ; the meta object is not versioned
-  // when the bucket is, as that would add an unneeded delete marker
-  int r = meta_obj->delete_object(this, y, true /* prevent versioning */);
```

Comment:
before this moved to `RGWCompleteMultipart::complete()`, it was only being called on success. but `complete()` is called on errors too, so the change caused a regression in https://tracker.ceph.com/issues/65746